esxi/tests/test_unit.py

```diff
@@ -0,0 +1,103 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import logging
+
+import pytest
+from mock import MagicMock
+from pyVmomi import vim, vmodl
+
+from datadog_checks.esxi import EsxiCheck
+
+
+def test_emits_critical_service_check_when_service_is_down(dd_run_check, aggregator, instance, caplog):
+    check = EsxiCheck('esxi', {}, [instance])
+    caplog.set_level(logging.WARNING)
+    dd_run_check(check)
+
+    aggregator.assert_metric('esxi.host.can_connect', value=0, tags=["esxi_url:localhost"])
+    assert "Cannot connect to ESXi host" in caplog.text
+
+
+@pytest.mark.usefixtures("service_instance")
+def test_esxi_none_properties_data(vcsim_instance, dd_run_check, aggregator, service_instance, caplog):
+    service_instance.content.propertyCollector.RetrievePropertiesEx = MagicMock(return_value=None)
+    check = EsxiCheck('esxi', {}, [vcsim_instance])
+    caplog.set_level(logging.WARNING)
+    dd_run_check(check)
+
+    assert "Did not retrieve any properties from the ESXi host." in caplog.text
+
+    base_tags = ["esxi_url:127.0.0.1:8989"]
+    aggregator.assert_metric("esxi.host.can_connect", 1, count=1, tags=base_tags)
+    aggregator.assert_all_metrics_covered()
```

Comment:
Why are we asserting this here? I guess in this test we should test that the properties we are not retrieving are not being reported

Reply:
We're asserting that the check stopped execution after the "Did not retrieve any properties from the ESXi host" log line and did not collect any performance metrics